core/src/operations/mod.rs

```diff
@@ -980,28 +839,131 @@ impl Deserializable for Operation {
+            OPCODE_FMPADD => Self::FmpAdd,
+            OPCODE_MLOAD => Self::MLoad,
+            OPCODE_SWAP => Self::Swap,
+            OPCODE_CALLER => Self::Caller,
+            OPCODE_MOVUP2 => Self::MovUp2,
+            OPCODE_MOVDN2 => Self::MovDn2,
+            OPCODE_MOVUP3 => Self::MovUp3,
+            OPCODE_MOVDN3 => Self::MovDn3,
+            OPCODE_ADVPOPW => Self::AdvPopW,
+            OPCODE_EXPACC => Self::Expacc,
+
+            OPCODE_MOVUP4 => Self::MovUp4,
+            OPCODE_MOVDN4 => Self::MovDn4,
+            OPCODE_MOVUP5 => Self::MovUp5,
+            OPCODE_MOVDN5 => Self::MovDn5,
+            OPCODE_MOVUP6 => Self::MovUp6,
+            OPCODE_MOVDN6 => Self::MovDn6,
+            OPCODE_MOVUP7 => Self::MovUp7,
+            OPCODE_MOVDN7 => Self::MovDn7,
+            OPCODE_SWAPW => Self::SwapW,
+            OPCODE_EXT2MUL => Self::Ext2Mul,
+            OPCODE_MOVUP8 => Self::MovUp8,
+            OPCODE_MOVDN8 => Self::MovDn8,
+            OPCODE_SWAPW2 => Self::SwapW2,
+            OPCODE_SWAPW3 => Self::SwapW3,
+            OPCODE_SWAPDW => Self::SwapDW,
+
+            OPCODE_ASSERT => {
+                let value_le_bytes: [u8; 4] = source.read_array()?;
+                let value = u32::from_le_bytes(value_le_bytes);
+                Self::Assert(value)
+            }
+            OPCODE_EQ => Self::Eq,
+            OPCODE_ADD => Self::Add,
+            OPCODE_MUL => Self::Mul,
+            OPCODE_AND => Self::And,
+            OPCODE_OR => Self::Or,
+            OPCODE_U32AND => Self::U32and,
+            OPCODE_U32XOR => Self::U32xor,
+            OPCODE_FRIE2F4 => Self::FriE2F4,
+            OPCODE_DROP => Self::Drop,
+            OPCODE_CSWAP => Self::CSwap,
+            OPCODE_CSWAPW => Self::CSwapW,
+            OPCODE_MLOADW => Self::MLoadW,
+            OPCODE_MSTORE => Self::MStore,
+            OPCODE_MSTOREW => Self::MStoreW,
+            OPCODE_FMPUPDATE => Self::FmpUpdate,
+
+            OPCODE_PAD => Self::Pad,
+            OPCODE_DUP0 => Self::Dup0,
+            OPCODE_DUP1 => Self::Dup1,
+            OPCODE_DUP2 => Self::Dup2,
+            OPCODE_DUP3 => Self::Dup3,
+            OPCODE_DUP4 => Self::Dup4,
+            OPCODE_DUP5 => Self::Dup5,
+            OPCODE_DUP6 => Self::Dup6,
+            OPCODE_DUP7 => Self::Dup7,
+            OPCODE_DUP9 => Self::Dup9,
+            OPCODE_DUP11 => Self::Dup11,
+            OPCODE_DUP13 => Self::Dup13,
+            OPCODE_DUP15 => Self::Dup15,
+            OPCODE_ADVPOP => Self::AdvPop,
+            OPCODE_SDEPTH => Self::SDepth,
+            OPCODE_CLK => Self::Clk,
+
+            OPCODE_U32ADD => Self::U32add,
+            OPCODE_U32SUB => Self::U32sub,
+            OPCODE_U32MUL => Self::U32mul,
+            OPCODE_U32DIV => Self::U32div,
+            OPCODE_U32SPLIT => Self::U32split,
+            OPCODE_U32ASSERT2 => {
+                let value_le_bytes: [u8; 4] = source.read_array()?;
+                let value = u32::from_le_bytes(value_le_bytes);
```

Comment:
nit: I'd name this variable `err_code` (and similar comment for other assert/verify operations).

Reply:
fixed